src/Quota_Command.php

```diff
@@ -0,0 +1,420 @@
+	 *
+	 * * blog_id
+	 * * url
+	 * * quota
+	 * * quota_used
+	 * * quota_used_percent
+	 *
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Output the quota of blog 3
+	 *     $ wp quota get 3
+	 *     +---------+--------------------------------------------------------+--------------+----------------+---------------------+
+	 *     | blog_id | url                                                    | quota        | quota_used     | quota_used_percent  |
+	 *     +---------+--------------------------------------------------------+--------------+----------------+---------------------+
+	 *     | 3       | https://dev-site.local/subsite3                        | 10000        | 9850           | 98.50               |
+	 *     +---------+--------------------------------------------------------+--------------+----------------+---------------------+
+	 *
+	 * @subcommand get
+	 */
+	public function get( $args, $assoc_args ) {
+		if ( ! is_multisite() ) {
+			WP_CLI::error( 'This is not a multisite installation.' );
+		}
+
+		if ( empty( $args ) ) {
+			$blog_id = get_current_blog_id();
+		} else {
+			$blog_id = $args[0] ?? 0;
+			$blog    = get_blog_details( $blog_id );
+			if ( ! $blog ) {
+				WP_CLI::error( 'Site not found.' );
+			}
+		}
+
+		$this->display_formatted_items( $assoc_args, $blog_id );
+
+	}
+
+	/**
+	 * Sets the quota for the chosen site to the given value
+	 *
+	 * ## OPTIONS
+	 *
+	 * [<id>]
+	 * : The id of the site to set the quota.
+	 *
+	 * [<quota-in-mb>]
+	 * : New quota value in mb
+	 *
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Set the quota of blog 2
+	 *     $ wp quota set 2 100000
+	 *
+	 *     Quota is now 10000 MB for subsite2
+	 *
+	 * @subcommand set
+	 */
+	public function set( $args, $assoc_args ) {
+		if ( ! is_multisite() ) {
+			WP_CLI::error( 'This is not a multisite installation.' );
+		}
+
+		if ( empty( $args ) ) {
+			WP_CLI::error( 'Need to specify a blog id.' );
+		}
+
+		if ( 2 == count( $args ) ) {
+			$blog_id         = (int) ( $args[0] ?? 0 );
+			$new_quota_in_mb = (int) ( $args[1] ?? 0 );
+		}
+
+		if ( 1 == count( $args ) ) {
+			$blog_id         = get_current_blog_id();
+			$new_quota_in_mb = (int) ( $args[0] ?? 0 );
+		}
+
+		if ( $blog_id ) {
+			$blog = get_blog_details( $blog_id );
+		}
+
+		if ( ! $blog ) {
+			WP_CLI::error( 'Site not found.' );
+		}
+
+		switch_to_blog( $blog_id );
+
+		$this->set_new_quota( $new_quota_in_mb, $blog );
+
+		restore_current_blog();
+
+	}
+
+	/**
+	 * Set the new blog quota
+	 *
+	 * @param $new_quota_in_mb
+	 * @param $blog
+	 */
+	private function set_new_quota( $new_quota_in_mb, $blog ) {
+		$global_blog_upload_max_space = get_network_option( get_current_network_id(), 'blog_upload_space' );
+		if ( $new_quota_in_mb != (int) $global_blog_upload_max_space ) {
+			$site_url = trailingslashit( $blog->siteurl );
+			update_option( 'blog_upload_space', $new_quota_in_mb );
+			WP_CLI::success( "Quota is now {$new_quota_in_mb} MB for {$site_url}." );
+		}
```

Comment:
Hier fehlt der else case. Die lokale `blog_upload_space` Option muss dann gelöscht werden. Aktuell kann man das Quota Limit nicht zurück auf den globalen Default setzen.

Reply:
Ich würde das sogar anders machen: Den Check auf `new_quota == global_quota` um das `update_option' legen und im Positivfall dann `delete_option `ausführen, statt `update_option`

Reply:
```suggestion
		$site_url = trailingslashit( $blog->siteurl );
		if ( $new_quota_in_mb != (int) $global_blog_upload_max_space ) {
			update_option( 'blog_upload_space', $new_quota_in_mb );
		} else {
			delete_option( 'blog_upload_space');
		}
		WP_CLI::success( "Quota is now {$new_quota_in_mb} MB for {$site_url}." );

```

Reply:
Habe ich nun geändert. Allerdings habe ich den Code so direkt aus dem "mip-customisations"-Projekt entnommen. Vielleicht dort das auch mal ändern?
#https://github.com/Hive-IT-GmbH/mip-customisations/blob/b6e4e4e76a4c8d8327212238114c319582004d33/mip-admin-quota-editor.php#L106-L114